File: materias 3 periodo ple/programacao orientada ao objeto/trabalhos praticos/tp-2/SistemaOficina/src/sistemaoficina/Servicos/Item.java
package sistemaoficina.Servicos;

public class Item implements Cloneable{
	private String nome;
	private double valorTotal;
	private int quantidade;
	public int codeItem;
	
	//Constructors
	public Item(){}
	
	public Item(String nome, double valorTotal, int quantidade, int codeItem){
            setNome(nome);
            setValorTotal(valorTotal);
            setQuantidade(quantidade);
            setCodeItem(codeItem);
		
	}
	
	//Gets e setters
	
	//Get e set nome
	public void setNome(String nome){
            this.nome = nome;
	}
	public String getNome(){
            return nome;
	}
	
	//Get e set Valor total
	public void setValorTotal(double valorTotal){
            this.valorTotal = valorTotal;
	}
	public double getValorTotal(){
            setValorTotal(valorTotal * quantidade);
            return (valorTotal * quantidade);
	}
	
	//Get e set quantidade
	public void setQuantidade(int quantidade){
            this.quantidade = quantidade;
	}

	public int getQuatidade(){
            return quantidade;
	}
	
	//Get e set codeItem
	public void setCodeItem(int codeItem){
            this.codeItem = codeItem;
	}

	public int getCodeItem(){
            return codeItem;
	}
	
	//Metodo clone
        
        
        @Override
        public Object clone(){
            try{                
                return super.clone();
                
            }catch(CloneNotSupportedException e){
                return null;
            }
        }
		

}

File: materias 3 periodo ple/programacao orientada ao objeto/trabalhos praticos/tp-1/Codigo/Cpp/Item.cpp
#include "../Hpp/Item.hpp"


Item :: Item(string nome, double preco, int quantidade, int codeItem) : nome(nome), preco(preco), quantidade(quantidade), codeItem(codeItem){}

Item :: ~Item(){}

// Gets

string Item :: getNome(){
    return nome;
}

double Item :: getPreco(){
    return preco;
}

int Item :: getQuantidade(){
    return quantidade;
}

double Item :: ValorTotal(){
    return preco*quantidade;
}

// Sets

void Item :: setNome(string nome){
    this->nome = nome;
}

void Item :: setPreco(double preco){
    this->preco = preco;
}

void Item :: setQuantidade(int quantidade){
    this->quantidade = quantidade;

}
int Item :: getCodeItem(){
    return codeItem;
}
    
void Item :: setCodeItem(int codeItem){
    this->codeItem = codeItem;
}  



